lib/WebService/GrowthBook.pm

```diff
@@ -0,0 +1,174 @@
+package WebService::GrowthBook;
+# ABSTRACT: ...
+
+use strict;
+use warnings;
+no indirect;
+use feature qw(state);
+use Object::Pad;
+use JSON::MaybeUTF8 qw(decode_json_text);
+use Scalar::Util qw(blessed);
+use Log::Any qw($log);
+use WebService::GrowthBook::FeatureRepository;
+use WebService::GrowthBook::Feature;
+use WebService::GrowthBook::FeatureResult;
+use WebService::GrowthBook::InMemoryFeatureCache;
+
+our $VERSION = '0.001';
+
+=head1 NAME
+
+WebService::GrowthBook - sdk of growthbook
+
+=head1 SYNOPSIS
+
+    use WebService::GrowthBook;
+    my $instance = WebService::GrowthBook->new(client_key => 'my key');
+    $instance->load_features;
+    if($instance->is_on('feature_name')){
+        # do something
+    }
+    else {
+        # do something else
+    }
+    my $string_feature = $instance->get_feature_value('string_feature');
+    my $number_feature = $instance->get_feature_value('number_feature');
+    # get decoded json
+    my $json_feature = $instance->get_feature_value('json_feature');
+
+=head1 DESCRIPTION
+
+    This module is a sdk of growthbook, it provides a simple way to use growthbook features.
+
+=cut
+
+# singletons
+
+class WebService::GrowthBook {
+    field $url :param //= 'https://api.growthbook.io/api/v1';
+    field $client_key :param;
+    field $features :param //= {};
+    field $cache_ttl :param //= 60;
+    field $cache //= WebService::GrowthBook::InMemoryFeatureCache->singleton;
```

Comment:
do we need this cache object here ? From my understanding The FeatureRepository seems to be abstracting the cache from the the Growthbook class Object ?

Reply:
This parameter is used to set cache module in FeatureRepository. Without this one we cannot custom cache object in FeatureRepository.